src/otaclient/_utils.py

```diff
@@ -41,26 +41,53 @@ def wait_and_log(
     check_interval: int = 2,
     log_interval: int = 30,
     log_func: Callable[[str], None] = logger.info,
-) -> None:
-    """Wait for <flag> until it is set while print a log every <log_interval>."""
+    timeout: int | None = None,
```

Comment:
add timeout for `wait_and_log`.
The motivation is to set timeout for sub ecu waiting in client update.